### PR TITLE
@craigspaeth - Forced login inquiry QA

### LINKED ISSUE #404: Forced Signup QA
* [x] Mobile: When trying to create an account I got an error msg: An error occurred, then user already exists. The `created_account` event does fire upond the first attempt. 
    * https://cl.ly/3D0k100x0H1x
    * https://cl.ly/463i2z2N2i3R
* [ ] Mobile: Was able to create an inquiry without logging in
* [x] Mobile: validation turned location field into undefined and any changes to the field would revert to undefined
    * https://cl.ly/0m1K1c0Z1Z0e
* [x] Desktop: location field errors out before any interactions: 
    * https://cl.ly/1a3N051Z3G23
* [x] Desktop: tried to inquire on a pre-qual flow work and used an email with an artsy account. got an error upon sending inquiry (Cab: "Admins are not allowed to place inquiries")
    * https://cl.ly/1n172U043a0W

### PR DESCRIPTION
**Pull request by broskoski**
Closes #404 

Kind of hard to test this locally, but there is an actual CORS issue happening. Microgravity is proxy-ing Force but then requesting to log_in using the actual Force url and getting a `No 'Access-Control-Allow-Origin'` error. Moving the `cors` middleware to before `artsy-passport` should fix this issue.

But..... now getting the dreaded `asset is not a function` when trying to hit that route from MG, I can't tell if this is just my local setup or not. When you get a second (ha!), mind pulling this down and seeing if you can replicate locally?

- Start mg in staging 
- Start force in staging
- Visit an artwork inquiry page (I used http://localhost:3333/inquiry/jon-cattapan-september-session ) in the mobile mode of your browser
- Try to place an inquiry with a new email and create an account (hitting `/log_in` is what I couldn't get past)

### CODE REVIEW

**Comment by craigspaeth** [comment]
Ok, so I dug into it and I'm seeing a 302 response when POSTing to /sign_up... which seems to be due to how we try and [redirect back](https://github.com/artsy/artsy-passport/blob/master/lib/app/index.coffee#L45) after signing up. Changing `req.xhr` to `req.accepts 'application/json'` seems to get past this issue, but then it runs into a CSRF issue.

The `asset is not a function` issue could be stifling these errors—I also saw a `User already exists` surfaced from the API. I went ahead and made some small changes to [not stifle those errors](https://github.com/artsy/artsy-error-handler/pull/14) in our error handler.

However, given I'm working towards merging MG + Force and this seems to be a bit of a rabbit hole b/c of the  proxying approach—maybe we should go ahead and pair on refactoring this to not use the proxying and do something more sane. Feel free to put a screen hero session on the calendar.

**Comment by broskoski** [comment]
Thanks for looking @craigspaeth. I'm gonna give it one more little shot with the updates you made to the error handler. But you're right, we should refactor auth regardless.

**Comment by broskoski** [comment]
OK @craigspaeth , you're correct, the CSRF issue is a blocker. It's probably worth merging this anyway, even though it doesn't fix the main issue it fixes a few minor ones (and updates artsy-error-handler). 

It is strange though, the  `csrf` cookie value is identical on both local force and local microgravity, I'm sure I'm missing something but I don't see how the token would be invalid.

File: lib/setup.coffee [hunk 1]
```diff
@@ -144,6 +144,12 @@ module.exports = (app) ->
     # secureProxy just sets secure=true
     secureProxy: "production" is NODE_ENV or "staging" is NODE_ENV
 
+  # We want the user to be able to log-in to force via the microgravity subdomain
+  # the initial use case being the professional buyer application
+  # (this is specific to responsive pages that require log-in)
+  console.log('[APP_URL, MOBILE_URL]', [APP_URL, MOBILE_URL])
```
Comment (craigspaeth) [inline lib/setup.coffee]: Stray log here, otherwise happy to merge 😁

**Comment by craigspaeth** [comment]
Sounds good. Feel free to merge when you remove that stray log and it's 💚

**Pull request merged**